unit test TestSELinuxMU fails on OpenIndiana

### Summary

I'm trying to package ansible version 2.14.4 for OpenIndiana and when I run tests some TestSELinuxMU tests fails.  Please note that OpenIndiana (a Solaris clone) have no selinux support.

### Issue Type

Bug Report

### Component Name

test_selinux.py

### Ansible Version

```console
$ ansible --version
ansible [core 2.14.4]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/marcel/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/vendor-packages/ansible
  ansible collection location = /home/marcel/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.16 (main, Feb 19 2023, 15:42:40) [GCC 10.4.0] (/usr/bin/python3.9)
  jinja version = 3.0.3
  libyaml = True
$
```


### Configuration

```console
NA.  Tests are run directly in the source directory after the source tarball is unpacked.
```


### OS / Environment

OpenIndiana

### Steps to Reproduce

```
$ bin/ansible-test units --python 3.9 --python-interpreter /usr/bin/python3.9 --local --color no --verbose
```


### Expected Results

All tests pass; selinux tests either pass, or are skipped.

### Actual Results

```console
Following tests fails:

FAILED test/units/module_utils/basic/test_selinux.py::TestSELinuxMU::test_selinux_mls_enabled
FAILED test/units/module_utils/basic/test_selinux.py::TestSELinuxMU::test_selinux_context
FAILED test/units/module_utils/basic/test_selinux.py::TestSELinuxMU::test_selinux_enabled
FAILED test/units/module_utils/basic/test_selinux.py::TestSELinuxMU::test_set_context_if_different
FAILED test/units/module_utils/basic/test_selinux.py::TestSELinuxMU::test_selinux_default_context
```

All failures are similar:
```
_________________ TestSELinuxMU.test_set_context_if_different __________________
[gw0] sunos5 -- Python 3.9.16 /usr/bin/python3.9
thing = <module 'ansible.module_utils.compat' from '/tmp/ansible-test-0qd9w7ab/ansible/module_utils/compat/__init__.py'>
comp = 'selinux', import_path = 'ansible.module_utils.compat.selinux'

    def _dot_lookup(thing, comp, import_path):
        try:
>           return getattr(thing, comp)
E           AttributeError: module 'ansible.module_utils.compat' has no attribute 'selinux'

/usr/lib/python3.9/unittest/mock.py:1226: AttributeError

During handling of the above exception, another exception occurred:

    from __future__ import (absolute_import, division, print_function)
    __metaclass__ = type

    import os
    import sys

    from ansible.module_utils.common.text.converters import to_native, to_bytes
    from ctypes import CDLL, c_char_p, c_int, byref, POINTER, get_errno

    try:
>       _selinux_lib = CDLL('libselinux.so.1', use_errno=True)

/tmp/ansible-test-0qd9w7ab/ansible/module_utils/compat/selinux.py:14:

...
...
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

```
_________________ TestSELinuxMU.test_set_context_if_different __________________
[gw3] sunos5 -- Python 3.9.16 /usr/bin/python3.9
thing = <module 'ansible.module_utils.compat' from '/tmp/ansible-test-ou71e3hn/ansible/module_utils/compat/__init__.py'>
comp = 'selinux', import_path = 'ansible.module_utils.compat.selinux'

    def _dot_lookup(thing, comp, import_path):
        try:
>           return getattr(thing, comp)
E           AttributeError: module 'ansible.module_utils.compat' has no attribute 'selinux'

/usr/lib/python3.9/unittest/mock.py:1226: AttributeError

During handling of the above exception, another exception occurred:

    from __future__ import (absolute_import, division, print_function)
    __metaclass__ = type
    
    import os
    import sys
    
    from ansible.module_utils.common.text.converters import to_native, to_bytes
    from ctypes import CDLL, c_char_p, c_int, byref, POINTER, get_errno
    
    try:
>       _selinux_lib = CDLL('libselinux.so.1', use_errno=True)

/tmp/ansible-test-ou71e3hn/ansible/module_utils/compat/selinux.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <CDLL 'libselinux.so.1', handle 0 at 0x7fffa702ba90>
name = 'libselinux.so.1', mode = 0, handle = None, use_errno = True
use_last_error = False, winmode = None

    def __init__(self, name, mode=DEFAULT_MODE, handle=None,
                 use_errno=False,
                 use_last_error=False,
                 winmode=None):
        self._name = name
        flags = self._func_flags_
        if use_errno:
            flags |= _FUNCFLAG_USE_ERRNO
        if use_last_error:
            flags |= _FUNCFLAG_USE_LASTERROR
        if _sys.platform.startswith("aix"):
            """When the name contains ".a(" and ends with ")",
               e.g., "libFOO.a(libFOO.so)" - this is taken to be an
               archive(member) syntax for dlopen(), and the mode is adjusted.
               Otherwise, name is presented to dlopen() as a file argument.
            """
            if name and name.endswith(")") and ".a(" in name:
                mode |= ( _os.RTLD_MEMBER | _os.RTLD_NOW )
        if _os.name == "nt":
            if winmode is not None:
                mode = winmode
            else:
                import nt
                mode = nt._LOAD_LIBRARY_SEARCH_DEFAULT_DIRS
                if '/' in name or '\\' in name:
                    self._name = nt._getfullpathname(self._name)
                    mode |= nt._LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR
    
        class _FuncPtr(_CFuncPtr):
            _flags_ = flags
            _restype_ = self._func_restype_
        self._FuncPtr = _FuncPtr
    
        if handle is None:
>           self._handle = _dlopen(self._name, mode)
E           OSError: ld.so.1: python3.9: fatal: libselinux.so.1: open failed: No such file or directory

/usr/lib/python3.9/ctypes/__init__.py:374: OSError

During handling of the above exception, another exception occurred:

self = <units.module_utils.basic.test_selinux.TestSELinuxMU object at 0x7fffa7dceb80>

    def test_set_context_if_different(self):
        am = no_args_module(selinux_enabled=False)
        assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], True) is True
        assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], False) is False
    
        am = no_args_module(selinux_enabled=True, selinux_mls_enabled=True)
        am.selinux_context = lambda path: ['bar_u', 'bar_r', None, None]
        am.is_special_selinux_path = lambda path: (False, None)
    
>       with patch('ansible.module_utils.compat.selinux.lsetfilecon', return_value=0) as m:

test/units/module_utils/basic/test_selinux.py:169: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.9/unittest/mock.py:1388: in __enter__
    self.target = self.getter()
/usr/lib/python3.9/unittest/mock.py:1563: in <lambda>
    getter = lambda: _importer(target)
/usr/lib/python3.9/unittest/mock.py:1239: in _importer
    thing = _dot_lookup(thing, comp, import_path)
/usr/lib/python3.9/unittest/mock.py:1228: in _dot_lookup
    __import__(import_path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    from __future__ import (absolute_import, division, print_function)
    __metaclass__ = type
    
    import os
    import sys
    
    from ansible.module_utils.common.text.converters import to_native, to_bytes
    from ctypes import CDLL, c_char_p, c_int, byref, POINTER, get_errno
    
    try:
        _selinux_lib = CDLL('libselinux.so.1', use_errno=True)
    except OSError:
>       raise ImportError('unable to load libselinux.so')
E       ImportError: unable to load libselinux.so

/tmp/ansible-test-ou71e3hn/ansible/module_utils/compat/selinux.py:16: ImportError
```

--- a/test/units/module_utils/basic/test_selinux.py
+++ b/test/units/module_utils/basic/test_selinux.py
@@ -183,7 +183,7 @@ class TestSELinuxMU:
         am.is_special_selinux_path = lambda path: (False, None)
 
         with patch.object(basic, 'selinux', create=True) as selinux:
-            selinux.lsetfilecon.return_value=0
+            selinux.lsetfilecon.return_value = 0
             assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], False) is True
             selinux.lsetfilecon.assert_called_with('/path/to/file', 'foo_u:foo_r:foo_t:s0')
             selinux.lsetfilecon.reset_mock()

--- a/test/units/module_utils/basic/test_selinux.py
+++ b/test/units/module_utils/basic/test_selinux.py
@@ -43,16 +43,21 @@ class TestSELinuxMU:
         with patch.object(basic, 'HAVE_SELINUX', False):
             assert no_args_module().selinux_enabled() is False
 
-        # test selinux present/not-enabled
-        disabled_mod = no_args_module()
-        with patch('ansible.module_utils.compat.selinux.is_selinux_enabled', return_value=0):
-            assert disabled_mod.selinux_enabled() is False
+            # test selinux present/not-enabled
+            disabled_mod = no_args_module()
+            with patch.object(basic, 'selinux', create=True) as selinux:
+                selinux.is_selinux_enabled.return_value = 0
+                assert disabled_mod.selinux_enabled() is False
+
         # ensure value is cached (same answer after unpatching)
         assert disabled_mod.selinux_enabled() is False
+
         # and present / enabled
-        enabled_mod = no_args_module()
-        with patch('ansible.module_utils.compat.selinux.is_selinux_enabled', return_value=1):
-            assert enabled_mod.selinux_enabled() is True
+        with patch.object(basic, 'HAVE_SELINUX', True):
+            enabled_mod = no_args_module()
+            with patch.object(basic, 'selinux', create=True) as selinux:
+                selinux.is_selinux_enabled.return_value = 1
+                assert enabled_mod.selinux_enabled() is True
         # ensure value is cached (same answer after unpatching)
         assert enabled_mod.selinux_enabled() is True
 
@@ -60,12 +65,16 @@ class TestSELinuxMU:
         # selinux unavailable, should return false
         with patch.object(basic, 'HAVE_SELINUX', False):
             assert no_args_module().selinux_mls_enabled() is False
-        # selinux disabled, should return false
-        with patch('ansible.module_utils.compat.selinux.is_selinux_mls_enabled', return_value=0):
-            assert no_args_module(selinux_enabled=False).selinux_mls_enabled() is False
-        # selinux enabled, should pass through the value of is_selinux_mls_enabled
-        with patch('ansible.module_utils.compat.selinux.is_selinux_mls_enabled', return_value=1):
-            assert no_args_module(selinux_enabled=True).selinux_mls_enabled() is True
+            # selinux disabled, should return false
+            with patch.object(basic, 'selinux', create=True) as selinux:
+                selinux.is_selinux_mls_enabled.return_value = 0
+                assert no_args_module(selinux_enabled=False).selinux_mls_enabled() is False
+
+        with patch.object(basic, 'HAVE_SELINUX', True):
+            # selinux enabled, should pass through the value of is_selinux_mls_enabled
+            with patch.object(basic, 'selinux', create=True) as selinux:
+                selinux.is_selinux_mls_enabled.return_value = 1
+                assert no_args_module(selinux_enabled=True).selinux_mls_enabled() is True
 
     def test_selinux_initial_context(self):
         # selinux missing/disabled/enabled sans MLS is 3-element None
@@ -80,16 +89,19 @@ class TestSELinuxMU:
             assert no_args_module().selinux_default_context(path='/foo/bar') == [None, None, None]
 
         am = no_args_module(selinux_enabled=True, selinux_mls_enabled=True)
-        # matchpathcon success
-        with patch('ansible.module_utils.compat.selinux.matchpathcon', return_value=[0, 'unconfined_u:object_r:default_t:s0']):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            # matchpathcon success
+            selinux.matchpathcon.return_value = [0, 'unconfined_u:object_r:default_t:s0']
             assert am.selinux_default_context(path='/foo/bar') == ['unconfined_u', 'object_r', 'default_t', 's0']
 
-        # matchpathcon fail (return initial context value)
-        with patch('ansible.module_utils.compat.selinux.matchpathcon', return_value=[-1, '']):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            # matchpathcon fail (return initial context value)
+            selinux.matchpathcon.return_value = [-1, '']
             assert am.selinux_default_context(path='/foo/bar') == [None, None, None, None]
 
-        # matchpathcon OSError
-        with patch('ansible.module_utils.compat.selinux.matchpathcon', side_effect=OSError):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            # matchpathcon OSError
+            selinux.matchpathcon.side_effect = OSError
             assert am.selinux_default_context(path='/foo/bar') == [None, None, None, None]
 
     def test_selinux_context(self):
@@ -99,19 +111,23 @@ class TestSELinuxMU:
 
         am = no_args_module(selinux_enabled=True, selinux_mls_enabled=True)
         # lgetfilecon_raw passthru
-        with patch('ansible.module_utils.compat.selinux.lgetfilecon_raw', return_value=[0, 'unconfined_u:object_r:default_t:s0']):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lgetfilecon_raw.return_value = [0, 'unconfined_u:object_r:default_t:s0']
             assert am.selinux_context(path='/foo/bar') == ['unconfined_u', 'object_r', 'default_t', 's0']
 
         # lgetfilecon_raw returned a failure
-        with patch('ansible.module_utils.compat.selinux.lgetfilecon_raw', return_value=[-1, '']):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lgetfilecon_raw.return_value = [-1, '']
             assert am.selinux_context(path='/foo/bar') == [None, None, None, None]
 
         # lgetfilecon_raw OSError (should bomb the module)
-        with patch('ansible.module_utils.compat.selinux.lgetfilecon_raw', side_effect=OSError(errno.ENOENT, 'NotFound')):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lgetfilecon_raw.side_effect = OSError(errno.ENOENT, 'NotFound')
             with pytest.raises(SystemExit):
                 am.selinux_context(path='/foo/bar')
 
-        with patch('ansible.module_utils.compat.selinux.lgetfilecon_raw', side_effect=OSError()):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lgetfilecon_raw.side_effect = OSError()
             with pytest.raises(SystemExit):
                 am.selinux_context(path='/foo/bar')
 
@@ -166,25 +182,29 @@ class TestSELinuxMU:
         am.selinux_context = lambda path: ['bar_u', 'bar_r', None, None]
         am.is_special_selinux_path = lambda path: (False, None)
 
-        with patch('ansible.module_utils.compat.selinux.lsetfilecon', return_value=0) as m:
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lsetfilecon.return_value=0
             assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], False) is True
-            m.assert_called_with('/path/to/file', 'foo_u:foo_r:foo_t:s0')
-            m.reset_mock()
+            selinux.lsetfilecon.assert_called_with('/path/to/file', 'foo_u:foo_r:foo_t:s0')
+            selinux.lsetfilecon.reset_mock()
             am.check_mode = True
             assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], False) is True
-            assert not m.called
+            assert not selinux.lsetfilecon.called
             am.check_mode = False
 
-        with patch('ansible.module_utils.compat.selinux.lsetfilecon', return_value=1):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lsetfilecon.return_value = 1
             with pytest.raises(SystemExit):
                 am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], True)
 
-        with patch('ansible.module_utils.compat.selinux.lsetfilecon', side_effect=OSError):
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lsetfilecon.side_effect = OSError
             with pytest.raises(SystemExit):
                 am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], True)
 
         am.is_special_selinux_path = lambda path: (True, ['sp_u', 'sp_r', 'sp_t', 's0'])
 
-        with patch('ansible.module_utils.compat.selinux.lsetfilecon', return_value=0) as m:
+        with patch.object(basic, 'selinux', create=True) as selinux:
+            selinux.lsetfilecon.return_value = 0
             assert am.set_context_if_different('/path/to/file', ['foo_u', 'foo_r', 'foo_t', 's0'], False) is True
-            m.assert_called_with('/path/to/file', 'sp_u:sp_r:sp_t:s0')
+            selinux.lsetfilecon.assert_called_with('/path/to/file', 'sp_u:sp_r:sp_t:s0')

